### ISSUE TITLE
Support for H617A

### ISSUE THREAD

**Issue by pfothergill** [Beshelmek/govee_ble_lights#37]
Would you be able to add support for H617A and/or tell me another configuration that would work with H617A?

**Comment by ExR90** [comment]
I have same question. Using H6171 and I get on/off and brightness but the color and theme functions do not work.

**Comment by wilwizard1979** [comment]
Same here, but I have 2 LED Strips H617E.
when I change the color in the colorwheel nothing happens in both cases. 
In one room it is always blue and only the brightness and On-Off works.
In the other room the brightness will change the brightness and color of the strip.
It would be great if this would work :-)

**Comment by rickestrada** [comment]
Hi all, chiming in, same with me, I tried the other H617* options but can't get anything other than on/off to work. Appreciate the help!

**Comment by algrym** [comment]
I saw this in the [README](https://github.com/Beshelmek/govee_ble_lights/blob/master/README.md):

> Almost every Govee device has its own BLE message protocol. If you have an Android smartphone and your device is not supported, please contact me on [Telegram](https://t.me/Beshelmek).

Does anyone have an H617A, an Android phone (I don't), and a Telegram account that could contact the author?  I suspect this might be as easy as sniffing BLE traffic with your phone.

**Comment by sierralx** [comment]
I’m attempting to assist Vladimir, but he has halted the development of this integration.

**Comment by wilwizard1979** [comment]
I have H617A to do some tests.
How can I help?

**Comment by wilwizard1979** [comment]
> I saw this in the [README](https://github.com/Beshelmek/govee_ble_lights/blob/master/README.md):
> 
> > Almost every Govee device has its own BLE message protocol. If you have an Android smartphone and your device is not supported, please contact me on [Telegram](https://t.me/Beshelmek).
> 
> Does anyone have an H617A, an Android phone (I don't), and a Telegram account that could contact the author? I suspect this might be as easy as sniffing BLE traffic with your phone.

I have an H617A but I don´t have an Android neither a Telegram account (I can create though).
I could try to sniff the traffic, but I just need the first baby steps. Can I do that from a PC? Which software can I use?

**Comment by Rombond** [comment]
I got 2 H617A so i have forked a version with H617C and adapt it to work on A model.
https://github.com/Rombond/govee_ble_lights

**Comment by DennisFury** [comment]
> I got 2 H617A so i have forked a version with H617C and adapt it to work on A model. https://github.com/Rombond/h617a_govee_ble_lights

You are the real MVP-- thank you!

**Comment by ExR90** [comment]
> I got 2 H617A so i have forked a version with H617C and adapt it to work on A model. https://github.com/Rombond/h617a_govee_ble_lights

Just out of curiosity, how did you sleuth down the commands needed to get the support made? I had considered doing so, as the code looked simple enough, but the codes themselves was the question-mark for me that I did not have a solution for

**Comment by Rombond** [comment]
> > I got 2 H617A so i have forked a version with H617C and adapt it to work on A model. https://github.com/Rombond/h617a_govee_ble_lights
> 
> Just out of curiosity, how did you sleuth down the commands needed to get the support made? I had considered doing so, as the code looked simple enough, but the codes themselves was the question-mark for me that I did not have a solution for

I used [RGB-PC](https://github.com/ib0b/RGB-PC) to know which data is required for it to works.

**Comment by alexpax85** [comment]
> > > I got 2 H617A so i have forked a version with H617C and adapt it to work on A model. https://github.com/Rombond/h617a_govee_ble_lights
> > 
> > 
> > Just out of curiosity, how did you sleuth down the commands needed to get the support made? I had considered doing so, as the code looked simple enough, but the codes themselves was the question-mark for me that I did not have a solution for
> 
> I used [RGB-PC](https://github.com/ib0b/RGB-PC) to know which data is required for it to works.

Hi Rombond,
I was looking for a way to trigger some scenes/presets that I can see from the app. With your fork installed in HA I can see the selector and the names of the effects, but they're not applied at all... something wrong with my H617A? I've set up either API and BLE [two enties in the integration: 1) Govee API with correct API key inside; 2) Govee light with one entity to contro].

I feel close, but at the moment it is working as the main integration (on/off, brightness control, color selection...) with no scene or segment control :(

**Comment by Rombond** [comment]
@alexpax85 I don't use scenes but this [fork](https://github.com/ahmadelsayed97/h617a_govee_ble_lights) seems to have updated the json to add scenes (havn't tested it).

**Comment by alexpax85** [comment]
@Rombond thanks for your reply. I tried that one, I forgot to mention it, sorry... I can see plenty of automation names from the gui, but none of them works. My only goal would be to setup a scene in the app (using the segments or movement effects) and then being able to call it from HA...

**Comment by algrym** [comment]
I agree with @alexpax85 on the built-in dynamic "scenes."   I've tried the [fork](https://github.com/ahmadelsayed97/h617a_govee_ble_lights) you mentioned and unfortunately with the same results (the dynamic are listed, but no effect when applied.)

I've got a bluetooth sniffer working with Wireshark and was hoping for something obvious, but quickly got in over my head.

I'm pretty sure I could get captures of the commands sent from the app for the scenes and we could compare those against the data in the fork.

I'll see what I can figure out, but if I'm in the weeds please let me know.

**Comment by alexpax85** [comment]
@algrym thanks for the help! you are far more skilled than me, if I could help in any way I’ll definitely give a try. Just let me know. 

Looking at the pythons file I see a lot of empty scene parameters. If you can sniff the command of the first preset listed in the file (forest?, don’t remember) we could proceed step by step at least with the default effects. To apply the DYI scenes, it’s completely over my capacity 😅

**Comment by teh-hippo** [comment]
@algrym , any luck?  Been trying to make these things more reliable in HA, and via BLE proxies.  They feel particularly finicky, and keep throwing GATT errors.

**Comment by Laserology** [comment]
We now provide an up to date fork that combines multiple community patches, and should support H617A

https://github.com/Laserology/govee_ble_lights

**Comment by teh-hippo** [comment]
Giving it a try @Laserology but so far there are immediate improvements.  Will engage in that repo. Thanks for flagging it here!
